File: 总题库/75.SortColors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int numsLen = nums.size();
        if (numsLen < 2) return;
        int low = 0, high = numsLen - 1;
        int i = 0;
        while (i <= high) {
            if (nums[i] == 0) {
                swap(nums[i++], nums[low++]);
            }
            else if (nums[i] == 2) {
                swap(nums[i], nums[high--]);
            }
            else ++i;
        }
        return;
    }
};

File: 总题库/75.Sortcolors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if len(nums) < 2:
            return None
        
        low = 0
        high = len(nums) - 1
        index = 0
        while index <= high:
            if nums[index] == 0:
                nums[index], nums[low] = nums[low], nums[index]
                index += 1
                low += 1
            elif nums[index] == 2:
                nums[index], nums[high] = nums[high], nums[index]
                high -= 1
            else:
                index += 1
